templates/Procfile

```diff
@@ -1,2 +1,2 @@
-web: bundle exec unicorn -p $PORT -c ./config/unicorn.rb
+web: bundle exec rails server Puma -p ${PORT:-3000} -c ./config/puma.rb
```

Comment:
The problem with `-c ./config/puma.rb` is that the rails server command is expecting a rackup file there, e.g. `config.ru` [by default](https://github.com/rails/rails/blob/4-2-stable/railties/lib/rails/commands/server.rb#L117).

To do what you want, what you basically need to do is:

```ruby
require ::File.expand_path('../config/environment', __FILE__)
require "puma"
require "puma/configuration"
puma_config_file = ::File.expand_path('../config/puma.rb', __FILE__)
Puma::Configuration.new(config_file: puma_config_file).load
run Rails.application
```

which doesn't work, so we'd need to ask the puma peeps how to configure puma from a rackup file